core-foundation/src/dictionary.rs

```diff
@@ -146,6 +165,26 @@ impl<K, V> CFMutableDictionary<K, V> {
         result
     }
 
+    #[inline]
+    pub fn to_untyped(&self) -> CFMutableDictionary {
+        unsafe { CFMutableDictionary::wrap_under_get_rule(self.0) }
+    }
+
+    /// Returns the same dictionary, but with the types reset to void pointers.
+    /// Equal to `to_untyped`, but is faster since it does not increment the retain count.
+    #[inline]
+    pub fn into_untyped(self) -> CFMutableDictionary {
+        let reference = self.0;
+        mem::forget(self);
+        unsafe { CFMutableDictionary::wrap_under_create_rule(reference) }
+    }
+
+    /// Borrow self as an immutable dictionary.
+    #[inline]
+    pub fn as_immutable(&self) -> ItemRef<CFDictionary<K, V>> {
```

Comment:
As I wrote in the initial description above, trying to limit this usage so the mutable dictionary acts as borrowed (thus immutable) while the returned value exists might be overkill/false claims. Since it's so easy to go around it anyway. Anyone can just clone their mutable dictionary and make one of them immutable and still have a mutable version.

With the added addition of
```rust
 impl<K, V> CFPropertyListSubClass for ::dictionary::CFMutableDictionary<K, V> {} 
```
below (Which we have not yet debated if it's correct or not), they could also just do
```rust
let mut_dict = CFMutableDictionary::new();
let dict = mut_dict.to_CFPropertyList().downcast::<CFDictionary>().unwrap();
```
and that way also have an owned immutable dictionary be live at the same time as a mutable version pointing to the same underlying dictionary.